gcc/jit/jit-recording.h

```diff
@@ -1227,6 +1227,9 @@ class compound_type : public type
 		 string *name);
 
   string *get_name () const { return m_name; }
+  void set_name (const char *new_name) {
+    m_name = m_ctxt->new_string (new_name);
+  }
```

Comment:
You also want to change the name of types?
If so, please add a test for this.

Reply:
Good point.